internal/appsec/waf_test.go

```diff
@@ -495,6 +502,149 @@ func TestAPISecurity(t *testing.T) {
 	})
 }
 
+func prepareSQLDB(nbEntries int) (*sql.DB, error) {
+	const tables = `
+CREATE TABLE user (
+   id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+   name  text NOT NULL,
+   email text NOT NULL,
+   password text NOT NULL
+);
+CREATE TABLE product (
+   id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+   name  text NOT NULL,
+   category  text NOT NULL,
+   price  int NOT NULL
+);
+`
+	db, err := sqltrace.Open("sqlite3", ":memory:")
+	if err != nil {
+		log.Fatalln("unexpected sql.Open error:", err)
+	}
+	sqltrace.Register("sqlite3", &sqlite3.SQLiteDriver{})
+
+	if _, err := db.Exec(tables); err != nil {
+		return nil, err
+	}
+
+	for i := 0; i < nbEntries; i++ {
+		_, err := db.Exec(
+			"INSERT INTO user (name, email, password) VALUES (?, ?, ?)",
+			fmt.Sprintf("User#%d", i),
+			fmt.Sprintf("user%d@mail.com", i),
+			fmt.Sprintf("secret-password#%d", i))
+		if err != nil {
+			return nil, err
+		}
+
+		_, err = db.Exec(
+			"INSERT INTO product (name, category, price) VALUES (?, ?, ?)",
+			fmt.Sprintf("Product %d", i),
+			"sneaker",
+			rand.Intn(500))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return db, nil
+}
+
+func TestRASPSQLi(t *testing.T) {
+	t.Setenv("DD_APPSEC_RULES", "testdata/rasp.json")
+	appsec.Start()
+	defer appsec.Stop()
+
+	if !appsec.RASPEnabled() {
+		t.Skip("RASP needs to be enabled for this test")
+	}
+	db, err := prepareSQLDB(10)
+	require.NoError(t, err)
+
+	// Setup the http server
+	mux := httptrace.NewServeMux()
+	mux.HandleFunc("/query", func(w http.ResponseWriter, r *http.Request) {
+		// Subsequent spans inherit their parent from context.
+		q := r.URL.Query().Get("query")
+		rows, err := db.QueryContext(r.Context(), q)
+		if events.IsSecurityError(err) {
+			w.WriteHeader(403)
+			w.Write([]byte(err.Error()))
```

Comment:
This is done by the blocking action in appsec middleware, you can just return

Reply:
Good point